### PR TITLE
DFA-1999 User changes mobile phone number

### PR DESCRIPTION
**Pull request by JohnWattsGDS**
Update mobile number validation to work with throws pattern
Use new validator in route
Update template to use values provided in error handler
Add some sugar to the steps in feature tests
Add new feature for "your account"
Add code to implement feature - change the number in Cognito and then use Cognito to send verification code, verify code and then set mobile as verified.  Then save the updated number in DDB. Eugh.
Some IDs needed to be added to views or changed to continue instead of submit to be a little less confusing.

### CODE REVIEW

File: express/src/controllers/manage-account.ts [hunk 1]
```diff
@@ -201,4 +210,72 @@ export const changePassword = async function (req: Request, res: Response) {
     }
     req.session.updatedField = "password";
     res.redirect('/account');
+}
+
+export const showChangePhoneNumberForm = async function (req: Request, res: Response) {
+    res.render('account/change-phone-number.njk', {values: {mobileNumber: req.session.mobileNumber}});
```
Comment (CharlesIC) [inline express/src/controllers/manage-account.ts:216]:
```suggestion
    res.render('account/change-phone-number.njk');
```
Remove to not pre-fill the form

File: express/src/controllers/manage-account.ts [hunk 2]
```diff
@@ -201,4 +210,72 @@ export const changePassword = async function (req: Request, res: Response) {
...
+}
+
+export const processChangePhoneNumberForm = async function (req: Request, res: Response) {
+    let mobile = req.body.mobileNumber;
+    if (mobile === "") {
+        const errorMessages = new Map<string, string>();
+        errorMessages.set('mobileNumber', 'Enter your mobile number');
+        res.render('account/change-phone-number.njk', {errorMessages: errorMessages});
+        return;
+    }
```
Comment (CharlesIC) [inline express/src/controllers/manage-account.ts:226]:
```suggestion
```
Remove - we decided this error checking is no longer needed

File: express/src/controllers/sign-in.ts [hunk 3]
```diff
@@ -39,6 +39,10 @@ export const showLoginOtpMobile = async function (req: Request, res: Response) {
 export const processLoginOtpMobile = async function (req: Request, res: Response) {
 
     if (!req.session.cognitoUser?.Username || !req.body['sms-otp'] || !req.session.session) {
+        console.log("Missing parameter")
+        console.log(req.session.cognitoUser?.Username);
+        console.log(req.body['sms-otp']);
+        console.log(req.session.session);
```
Comment (CharlesIC) [inline express/src/controllers/sign-in.ts:45]: Delete

File: express/src/controllers/sign-in.ts [hunk 4]
```diff
@@ -119,6 +123,7 @@ export const processSignInForm = async function (req: Request, res: Response) {
         res.render('there-is-a-problem.njk');
         return;
     }
+    console.log(response);
```
Comment (CharlesIC) [inline express/src/controllers/sign-in.ts:126]: Delete

File: express/src/controllers/manage-account.ts (hunk 1 above)
Comment (JohnWattsGDS) [reply express/src/controllers/manage-account.ts:216]: 👍🏻

File: express/src/controllers/sign-in.ts (hunk 4 above)
Comment (JohnWattsGDS) [reply express/src/controllers/sign-in.ts]: 🔥

File: express/src/controllers/sign-in.ts (hunk 3 above)
Comment (JohnWattsGDS) [reply express/src/controllers/sign-in.ts:42]: 🔥

File: express/src/controllers/manage-account.ts (hunk 1 above)
Comment (JohnWattsGDS) [reply express/src/controllers/manage-account.ts]: 🔥

**Review by CharlesIC — APPROVED** [review]

**Pull request merged**